tests/unit_tests/test_simulator.py

```diff
@@ -0,0 +1,343 @@
+
+
+@pytest.fixture
+def arrayConfigData(tmp_test_directory):
+    return {
+        "dataDirectory": str(tmp_test_directory) + "/test-output",
+        "primary": "gamma",
+        "zenith": 20 * u.deg,
+        "azimuth": 0 * u.deg,
+        "viewcone": [0 * u.deg, 0 * u.deg],
+        # ArrayModel
+        "site": "North",
+        "layoutName": "1LST",
+        "modelVersion": "Prod5",
+        "default": {"LST": "1"},
+        "M-01": "FlashCam-D",
+    }
+
+
+@pytest.fixture
+def input_file_list():
+    return ['run1', 'abc_run22', 'def_run02_and']
+
+
+@pytest.fixture
+def showerConfigData():
+    return {
+        "dataDirectory": ".",
+        "site": "South",
+        "layoutName": "Prod5",
+        "runList": [3, 4],
+        "runRange": [6, 10],
+        "nshow": 10,
+        "primary": "gamma",
+        "erange": [100 * u.GeV, 1 * u.TeV],
+        "eslope": -2,
+        "zenith": 20 * u.deg,
+        "azimuth": 0 * u.deg,
+        "viewcone": 0 * u.deg,
+        "cscat": [10, 1500 * u.m, 0],
+    }
+
+
+@pytest.fixture
+def corsikaFile():
+    return "run1_proton_za20deg_azm0deg-North-1LST_trigger_rates.corsika.zst"
+
+
+@pytest.fixture
+def array_simulator(set_db, label, arrayConfigData):
+
+    arraySimulator = Simulator(
+        label=label, simulator='simtel', configData=arrayConfigData)
+    return arraySimulator
+
+
+@pytest.fixture
+def shower_simulator(set_db, label, showerConfigData):
+
+    showerSimulator = Simulator(
+        label=label, simulator='corsika', configData=showerConfigData)
+    return showerSimulator
+
+
+def test_set_file_locations(array_simulator, shower_simulator):
+
+    assert Path.exists(array_simulator._simtelSourcePath)
+    assert Path.exists(array_simulator._filesLocation)
+    assert Path.exists(array_simulator._outputDirectory)
+
+    assert Path.exists(shower_simulator._simtelSourcePath)
+    assert Path.exists(shower_simulator._filesLocation)
+    assert Path.exists(shower_simulator._outputDirectory)
+
+
+def test_guess_run_from_file(array_simulator):
+
+    assert array_simulator._guessRunFromFile("run12345_bla_ble") == 12345
+
+    # Invalid run number - returns 1
+    assert array_simulator._guessRunFromFile("run1test2_bla_ble") == 1
+
+    assert array_simulator._guessRunFromFile("abc-run12345_bla_ble") == 12345
+
+    assert array_simulator._guessRunFromFile("run10") == 10
+
+    assert array_simulator._guessRunFromFile("abc-ran12345_bla_ble") == 1
+
+    # TODO add test for 'Error creating output directory'
+    # (not sure how to test a failed mkdir
+
+
+def test_set_simulator(array_simulator, shower_simulator):
+
+    array_simulator._set_simulator('simtel')
+    assert array_simulator.simulator == 'simtel'
+
+    shower_simulator._set_simulator('corsika')
+    assert shower_simulator.simulator == 'corsika'
+
+    with pytest.raises(gen.InvalidConfigData):
+        shower_simulator._set_simulator('this_simulator_is_not_there')
+
+
+def test_loadConfigData(array_simulator):
```

Comment:
The names of the test functions are inconsistent. Everywhere else in the tests (and in some of the tests here) we only used snake_case for function names. Here you mixed snake_case with camelCase. I know it's annoying, but I suggest to keep our convention.